geomesa-hbase/geomesa-hbase-datastore/src/main/java/org/locationtech/geomesa/hbase/filters/KryoLazyDensityFilter.java

```diff
@@ -0,0 +1,359 @@
+        return writePointToResult((Point) sf.getAttribute(geomIndex), weight);
+    }
+
+    private Map.Entry<Tuple2<Integer, Integer>, Double> writeNonPoint(Geometry geom, Double weight) {
+        return writePointToResult(safeCentroid(geom), weight);
+    }
+
+    private Point safeCentroid(Geometry geom) {
+        Polygon p = geom.noPolygon();
+        Point centroid = p.getCentroid();
+        if (Double.isNaN(centroid.getCoordinate().x) || Double.isNaN(centroid.getCoordinate().y)) {
+            return p.getEnvelope().getCentroid();
+        } else {
+            return centroid;
+        }
+    }
+
+    private Map.Entry<Tuple2<Integer, Integer>, Double> writePointToResult(Point pt, Double weight) {
+        return writeSnappedPoint(gridSnap.i(pt.getX()), gridSnap.j(pt.getY()), weight);
+    }
+
+    private Map.Entry<Tuple2<Integer, Integer>, Double> writeSnappedPoint(int x, int y, Double weight) {
+        Tuple2 tuple = new Tuple2(x, y);
+        return new AbstractMap.SimpleEntry(tuple, weight);
+
+    }
+
+    private void writePointToResult(Coordinate pt, Double weight) {
+        writeSnappedPoint(gridSnap.i(pt.x), gridSnap.j(pt.y), weight);
+    }
+
+    private void writePointToResult(double x, double y, double weight) {
+        writeSnappedPoint(gridSnap.i(x), gridSnap.j(y), weight);
+    }
+
+    @Override
+    public ReturnCode filterKeyValue(Cell v) throws IOException {
+        byte[] encodedSF = CellUtil.cloneValue(v);
+        SimpleFeature sf = serializer.deserialize(encodedSF);
+        return ReturnCode.INCLUDE;
+    }
+
+    @Override
+    public Cell transformCell(Cell v) throws IOException {
+        byte[] row = CellUtil.cloneRow(v);
+        byte[] encodedSF = CellUtil.cloneValue(v);
+        SimpleFeature sf_old = serializer.deserialize(encodedSF);
+        Map.Entry<Tuple2<Integer, Integer>, Double> out;
+        if (isPoints()) {
+            out = writePoint(sf_old, weightFn(sf_old));
+        } else {
+            out = writeNonPoint((Geometry) sf_old.getDefaultGeometry(), weightFn(sf_old));
+        }
+        SimpleFeature sf = new ScalaSimpleFeature(sf_old.getID(), outputsft, null, null);
+        sf.setAttribute(0, serializeParameters(out.getKey()));
+        sf.setAttribute(1, out.getValue());
+        byte[] arr = output_serializer.serialize(sf);
+        Cell cell = CellUtil.createCell(row, arr);
+        return cell;
+    }
+
+    // TODO: Add static method to compute byte array from SFT and Filter.
+    @Override
+    public byte[] toByteArray() throws IOException {
+//        System.out.println("Serializing JLazyDensityFiler!");
+        return Bytes.add(getLengthArray(sftString), convertToBytes(hints));
+    }
+
+    private byte[] convertToBytes(Object object) throws IOException {
+        if (object == null) {
+            return Bytes.toBytes(0);
+        }
+        try (ByteArrayOutputStream bos = new ByteArrayOutputStream();
```

Comment:
using java serialization here is not something we should do... we should find another way to configure the filter. The accumulo implementation stores everything as key-value strings, we should be able to duplicate that setup here.